rpm/gen-fixtures-delta.sh

```diff
@@ -0,0 +1,160 @@
+
+Create <output_dir>. Use packages in <assets_dir> to generate DRPM fixture data for package <delta_package_name>
+and place the results into <output_dir>. A suggested name for <output_dir> is "zoo".
+
+EOF
+cat <<EOF
+<output_dir>
+    The directory into which generated fixtures are placed.
+<assets_dir>
+    The directory from which source material is read.
+ <delta_package_name>
+    The name of package which delta RPM will be generated for
+EOF
+}
+
+# Given a path to an RPM package, echo the package name.
+#
+# e.g. assets/walrus-baby-5.21-1.noarch.rpm → walrus-baby
+get_rpm_name() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    local end=`expr ${#parts[@]} - 2`
+    for ((i=0; i<${end}; i++)); do
+        rpm_name+="${parts[i]}"-
+    done
+    rpm_name=${rpm_name::-1}
+    echo "${rpm_name#*\!}"  # strip epoch
+}
+
+# Given a path to an RPM package, echo the package version.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → 5.21
+get_rpm_version() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    echo "${parts[-2]}"
+}
+
+# Given a pathto an RPM package, echo the package release.
+#
+# e.g. assets/yum-cron-3.4.3-10.fc16.noarch.rpm  → 10.fc16
+get_rpm_release() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    last_part=${parts[-1]}
+    pattern="${2}.rpm"
+
+    release=${last_part%.$pattern}
+    echo "${release}"
+}
+
+# Given a path to an RPM package, echo the package arch.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → noarch
+get_rpm_arch() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr . "\n"))
+    echo "${parts[-2]}"
+}
+
+#------------------------------------------------------------------------------
+# Business Logic
+#------------------------------------------------------------------------------
+
+# Fetch output_dir from user.
+if [ "$#" -lt 2 ]; then
+    echo 1>&2 'Error: Too few arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+elif [ "$#" -gt 3 ]; then
+    echo 1>&2 'Error: Too many arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+else
+    output_dir="$(realpath --canonicalize-missing "${1}")"
+    assets_dir="$(realpath "${2}")"
+    package_name="${3}"
+fi
+
+# Copy all RPMs from assets_dir to output_dir
+# Create directory 'drpms' for DRPMS
+mkdir -vp "${output_dir}"/drpms
+cp -t "${output_dir}" "${assets_dir}/"*.rpm
+
+# Filter and sort all required RPMs.
+rpms="${output_dir}"/"${package_name}"*.rpm
+IFS=$'\n' rpms=($(sort <<<"${rpms[*]}"))
+
+found_rpms=()
+for (( i=0; i<${#rpms[@]}; i++ ));
+do
+    current_package_name="$(get_rpm_name "${rpms[i]}")"
+    if [ "${current_package_name}" == "${package_name}" ]; then
+        found_rpms+=("${rpms[i]}")
+    fi
+done
+
+readonly num_needed=2
```

Comment:
This is a single-use variable. Interpolate its value into the code below and drop this variable, or use it in the string error message.